src/adapters/github.js

```diff
@@ -146,12 +146,15 @@ class GitHub extends PjaxAdapter {
       return cb();
     }
 
+    const branchOnMenu = $('.branch-select-menu .select-menu-item.selected').data('name') || '';
+
     // Get branch by inspecting URL or DOM, quite fragile so provide multiple fallbacks
     const branch =
       // Branch/tag/commit from URL
-      (isCodePage && typeId) ||
+      // Exclude edge case: branch name has slash '/'
+      (!branchOnMenu.includes('/') && isCodePage && typeId) ||
```

Comment:
Or maybe we should use `typeId` only when type is 'commit'?

Reply:
How about we only use `typeId` when it's a url is `commit`? For others (`tree` and `blob`), just use `branchOnMenu`? What is simpler to understand and more future proof?

The reason I don't quite love the fix in this PR is it combines both URL and DOM to make a determination. 2 volatile things are more volatile in combination :).